units/src/amount.rs

```diff
@@ -1324,6 +1419,30 @@ pub mod serde {
         fn ser_btc_opt<S: Serializer>(self, s: S) -> Result<S::Ok, S::Error>;
     }
 
+    struct DisplayFullError(ParseAmountError);
```

Comment:
In 7bf478373a343ca0a4a89d49225ab58a5f6eb196:

I guess this is a question for a followup PR, but do we want to make this a generic thing and use it in all of our serde impls?

Relatedly, could we replace our `write_err!` macro with a wrapper type like this that simply has different `Display` impls with `std` and non-`std`?

Reply:
> do we want to make this a generic thing and use it in all of our serde impls?

Yes, I think so. Honestly this is the kind of thing that really should've been in `core`.

> Relatedly, could we replace our `write_err!` macro with a wrapper type like this that simply has different `Display` impls with `std` and non-`std`?

I think that would just create more code, not less because we'd still need `format_args!` to supply the arguments.